In [1]:
import pandas as pd
from pandas_profiling import ProfileReport

In [2]:
raw_datafile = '../data/census_cleaned.csv'
df = pd.read_csv(raw_datafile)

In [3]:
df.head(16)

age         workclass   fnlgt     education  education-num  \
0    39         State-gov   77516     Bachelors             13   
1    50  Self-emp-not-inc   83311     Bachelors             13   
2    38           Private  215646       HS-grad              9   
3    53           Private  234721          11th              7   
4    28           Private  338409     Bachelors             13   
5    37           Private  284582       Masters             14   
6    49           Private  160187           9th              5   
7    52  Self-emp-not-inc  209642       HS-grad              9   
8    31           Private   45781       Masters             14   
9    42           Private  159449     Bachelors             13   
10   37           Private  280464  Some-college             10   
11   30         State-gov  141297     Bachelors             13   
12   23           Private  122272     Bachelors             13   
13   32           Private  205019    Assoc-acdm             12   
14   40           Private  121772     Assoc-voc             11   
15   34           Private  245487       7th-8th              4   

           marital-status         occupation   relationship  \
0           Never-married       Adm-clerical  Not-in-family   
1      Married-civ-spouse    Exec-managerial        Husband   
2                Divorced  Handlers-cleaners  Not-in-family   
3      Married-civ-spouse  Handlers-cleaners        Husband   
4      Married-civ-spouse     Prof-specialty           Wife   
5      Married-civ-spouse    Exec-managerial           Wife   
6   Married-spouse-absent      Other-service  Not-in-family   
7      Married-civ-spouse    Exec-managerial        Husband   
8           Never-married     Prof-specialty  Not-in-family   
9      Married-civ-spouse    Exec-managerial        Husband   
10     Married-civ-spouse    Exec-managerial        Husband   
11     Married-civ-spouse     Prof-specialty        Husband   
12          Never-married       Adm-clerical      Own-child   
13          Never-married              Sales  Not-in-family   
14     Married-civ-spouse       Craft-repair        Husband   
15     Married-civ-spouse   Transport-moving        Husband   

                  race     sex  capital-gain  capital-loss  hours-per-week  \
0                White    Male          2174             0              40   
1                White    Male             0             0              13   
2                White    Male             0             0              40   
3                Black    Male             0             0              40   
4                Black  Female             0             0              40   
5                White  Female             0             0              40   
6                Black  Female             0             0              16   
7                White    Male             0             0              45   
8                White  Female         14084             0              50   
9                White    Male          5178             0              40   
10               Black    Male             0             0              80   
11  Asian-Pac-Islander    Male             0             0              40   
12               White  Female             0             0              30   
13               Black    Male             0             0              50   
14  Asian-Pac-Islander    Male             0             0              40   
15  Amer-Indian-Eskimo    Male             0             0              45   

   native-country  salary  
0   United-States       0  
1   United-States       0  
2   United-States       0  
3   United-States       0  
4            Cuba       0  
5   United-States       0  
6         Jamaica       0  
7   United-States       1  
8   United-States       1  
9   United-States       1  
10  United-States       1  
11          India       1  
12  United-States       0  
13  United-States       0  
14            NaN       1  
15         Mexico       0

In [4]:
profile = ProfileReport(df, title='census dataset profile for EDA')

In [5]:
profile.to_widgets()

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

0 age: continuous.
1 workclass: Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked.
2 fnlwgt: continuous.
3 education: Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool.
4 education-num: continuous.
5 marital-status: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.
6 occupation: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.
7 relationship: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.
8 race: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.
9 sex: Female, Male.
10 capital-gain: continuous.
11 capital-loss: continuous.
12 hours-per-week: continuous.
13 native-country: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala

Count unknown values (NaN) in each column that contains strings

In [6]:
categorical_idx = (1, 3, 4, 5, 6, 7, 8, 9, 13)
df.isnull().sum()

age                  0
workclass         1836
fnlgt                0
education            0
education-num        0
marital-status       0
occupation        1843
relationship         0
race                 0
sex                  0
capital-gain         0
capital-loss         0
hours-per-week       0
native-country     583
salary               0
dtype: int64

Determine the fraction of samples in the dataset with either predicted value

In [7]:
df.salary.value_counts(normalize=True)

0    0.75919
1    0.24081
Name: salary, dtype: float64

The dataset is imbalanced, with around 76% of samples of class 0, and 24% of class 1.